### DISCUSSION TITLE
Question: How are zk notes supposed to be navigated?

### DISCUSSION THREAD

**Discussion by ndavd** [zk-org/zk#150] [Help] [answered]
Suppose I have a note which contains multiple links to other notes on the topic.
Ideally I would be able to open that note read its contents, click on a link and go to the respective note.

How can that be done?

**Comment by mickael-menu** [comment, marked as answer]
Usually you would pair `zk` with a Markdown editor (e.g. Obsidian, Zettlr) or any [LSP-compatible code editor](https://github.com/mickael-menu/zk/blob/main/docs/editors-integration.md) (e.g. VS Code)

If you prefer to stay in the terminal, many are using Neovim with `zk`. There's also a [dedicated Neovim plugin](https://github.com/mickael-menu/zk-nvim) to augment Neovim with `zk` features.

Alternatively, you can use tools to convert your notebook to a browsable website, such as [neuron](https://github.com/srid/neuron) or [gollum](https://github.com/gollum/gollum).

**Reply by ndavd** [reply]
I use neovim and I currently have that plugin installed. is there a way to follow the links using it?

**Reply by mickael-menu** [reply]
Yes, check out the [Example Mappings](https://github.com/mickael-menu/zk-nvim#example-mappings) section. 

Following links is handled with this mapping:

```lua
lua << EOF

vim.api.nvim_set_keymap("n", "<CR>", "<Cmd>lua vim.lsp.buf.definition()<CR>", { noremap=true })

EOF
```

**Reply by ndavd** [reply]
how cool thanks!
